grpc_opa/authorizer.go

```diff
@@ -343,3 +344,55 @@ func addObligations(ctx context.Context, opaResp OPAResponse) (context.Context,
 	}
 	return ctx, err
 }
+
+// NewAuthorizationConfigUnary returns new unary config for authorization requests
+func NewAuthorizationConfigUnary(application string, opts ...Option) *Config {
+	cfg := &Config{
+		address: opa_client.DefaultAddress,
+	}
+
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	if cfg.authorizer == nil {
+		log.Info("authorizers empty, using default authorizer")
+		cfg.authorizer = []Authorizer{NewDefaultAuthorizer(application, opts...)}
+	}
+
+	return cfg
+}
+
+// AffirmAuthorizationUnary makes an authz request to sidecar-OPA
+// If authorization allowed, error returned is nil,
+// and a new context is returned containing any obligations
+// which caller must further evaluate if required
+func AffirmAuthorizationUnary(ctx context.Context, cfg *Config, fullMethod string, grpcReq interface{}) (context.Context, error) {
```

Comment:
@daniel-garcia had a good comment: AffirmAuthorizationUnary should be a Config method.

Reply:
That would be fine too, you can then drop the 2nd arg.